tests/mocha/jso_deserialization_test.js

```diff
@@ -0,0 +1,157 @@
+  });
+
+  teardown(function() {
+    workspaceTeardown.call(this, this.workspace);
+    sharedTestTeardown.call(this);
+  });
+
+  suite('Events', function() {
+    test.skip('Finished loading', function() {
+      const state = {
+        'blocks': {
+          'blocks': [
+            {
+              'type': 'controls_if',
+              'id': 'testId',
+              'x': 42,
+              'y': 42
+            },
+          ]
+        }
+      };
+      Blockly.serialization.workspaces.load(state, this.workspace);
+      assertEventFired(
+          this.eventsFireStub,
+          Blockly.Events.FinishedLoading,
+          {},
+          this.workspace.id);
+    });
+
+    suite('Var create', function() {
+      test('Just var', function() {
+        const state = {
+          'variables': [
+            {
+              'name': 'test',
+              'id': 'testId',
+            }
+          ]
+        };
+        Blockly.serialization.workspaces.load(state, this.workspace);
+        assertEventFired(
+            this.eventsFireStub,
+            Blockly.Events.VarCreate,
+            {'varName': 'test', 'varId': 'testId', 'varType': ''},
+            this.workspace.id);
+      });
+
+      test('Var with block', function() {
+        const state = {
+          'variables': [
+            {
+              'name': 'test',
+              'id': 'testId',
+            }
+          ],
+          'blocks': {
+            'blocks': [
+              {
+                'type': 'variables_get',
+                'id': 'blockId',
+                'x': 42,
+                'y': 42,
+                'fields': {
+                  'VAR': 'testId'
+                }
+              },
+            ]
+          }
+        };
+        Blockly.serialization.workspaces.load(state, this.workspace);
+        const calls = this.eventsFireStub.getCalls();
+        const count = calls.reduce((acc, call) => {
+          if (call.args[0] instanceof Blockly.Events.VarCreate) {
+            return acc + 1;
+          }
+          return acc;
+        }, 0);
+        chai.assert.equal(count, 1);
+        assertEventFired(
+            this.eventsFireStub,
+            Blockly.Events.VarCreate,
+            {'varName': 'test', 'varId': 'testId', 'varType': ''},
+            this.workspace.id);
+      });
+    });
+
+    suite('Block create', function() {
+      test('No children', function() {
+        const state = {
+          'blocks': {
+            'blocks': [
+              {
+                'type': 'controls_if',
+                'id': 'testId',
+                'x': 42,
+                'y': 42
+              },
+            ]
+          }
+        };
+        Blockly.serialization.workspaces.load(state, this.workspace);
+        assertEventFired(
+            this.eventsFireStub,
+            Blockly.Events.BlockCreate,
+            {},
+            this.workspace.id,
+            'testId');
+      });
+
+      test('with children', function() {
```

Comment:
We don't really have a consistent style for tests imo... so feel free to ignore. But I personally like to add which behavior I expect to happen in the test so you can see at a glance. For some tests if it's really obvious then saying something like "Var creation causes var create event to be fired" might be obvious/redundant but for some of them I think "only top block event is fired" or something can make it obvious what should have happened, when a test fails. e.g. https://github.com/google/blockly/blame/master/tests/mocha/block_test.js#L1636

Reply:
I think that's a good suggestion! I think it would also be nice to have a styleguide for tests, or at least some more consistency in them haha.